File: android/src/main/java/com/getcapacitor/community/classes/events/StartedEvent.java
package com.getcapacitor.community.classes.events;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;

public class StartedEvent extends PeerIDEvent {

    @Nullable
    Boolean isStarted;

    public StartedEvent(@NonNull String peerID, @Nullable Boolean isStarted) {
        super(peerID);
        this.isStarted = isStarted;
    }

    @NonNull
    public JSObject toJSObject() {
        JSObject result = super.toJSObject();

        if (isStarted != null) result.put("isStarted", isStarted);

        return result;
    }
}

File: ios/Sources/BitchatPlugin/Classes/Events/StartedEvent.swift
import Foundation
import Capacitor

@objc public class StartedEvent: PeerIDEvent {
    let isStarted: Bool?

    init(_ peerID: String, _ isStarted: Bool?) {
        self.isStarted = isStarted

        super.init(peerID)
    }

    override public func toJSObject() -> JSObject {
        var result = super.toJSObject()

        if let isStarted = isStarted {
            result["isStarted"] = isStarted
        }

        return result
    }
}
